casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -158,11 +162,82 @@ class RuntimeManagerImpl[F[_]: Concurrent] private[rholang] (
       }
   }
 
-  private def withResetRuntime[R](hash: StateHash, block: Runtime[F] => F[R]) =
-    Sync[F].bracket(runtimeContainer.take) { runtime =>
-      val blakeHash = Blake2b256Hash.fromByteArray(hash.toByteArray)
-      runtime.space.reset(blakeHash).flatMap(_ => block(runtime))
-    }(runtime => runtimeContainer.put(runtime))
+  def computeBalance(start: StateHash)(user: ByteString): F[Long] =
+    withResetRuntime(start)(computeBalance(_)(user))
+
+  private def balanceQuerySource(user: ByteString): String =
+    s"""
+       | new rl(`rho:registry:lookup`), revAddressOps(`rho:rev:address`), revVaultCh in {
+       |   rl!(`rho:id:1o93uitkrjfubh43jt19owanuezhntag5wh74c6ur5feuotpi73q8z`, *revVaultCh)
```

Comment:
The `rho:id:1o93uitkrjfubh43jt19owanuezhntag5wh74c6ur5feuotpi73q8z` could be extracted to a constant, otherwise it's a magic string

Reply:
@lukasz-golebiewski My argument is the same as Artur's. Are you convinced?

Reply:
This contract id is already used in many places. I think it would make sense to introduce a `rho:contracts:vault` mapping instead of using an obscure hash in a separate PR